nilearn/interfaces/bids/query.py

```diff
@@ -1,7 +1,152 @@
 """Functions for working with BIDS datasets."""
+
+from __future__ import annotations
+
 import glob
+import json
 import os
 
+from pathlib import Path
+from warnings import warn
+
+def _get_metadata_from_bids(field,
+                            json_files,
+                            bids_path=None,):
+    """Get a metadata field from a BIDS json sidecar files.
+
+    This assumes that all the json files in the list have the same value
+    for that field,
+    hence the metadata is read only from the first json file in the list.
+
+    Parameters
+    ----------
+    field : :obj:`str`
+        Name of the field to be read. For example 'RepetitionTime'.
+
+    json_files : :obj:`list` of :obj:`str`
+        List of path to json files, for example returned by get_bids_files.
+
+    dataset_type : :obj:`str`
+        String to describe the type of dataset, 
+        for example either 'raw' or 'derivatives'.
+
+    Returns
+    -------
+    float or None
+        value of the field or None if the field is not found.
+    """
+    if json_files:
+        assert (
+                isinstance(json_files, list) 
+                and isinstance(json_files[0], (Path, str))
+                )
+        with open(json_files[0], 'r') as f:
+            specs = json.load(f)
+        value = specs.get(field)
+        if value is not None:
+            return value
+        else:
+            warn(f"'{field}' not found in file {json_files[0]}.")
+    else:
+        msg_suffix = f" in {bids_path}" if bids_path else ""
+        warn(f'No bold.json found in BIDS folder{msg_suffix}.')
```

Comment:
Should be a warning or an error ?

Reply:
I think I would stick to warning here: see my new comment in the PR.


https://github.com/nilearn/nilearn/pull/3605#issuecomment-1490567765

I think the errors should be thrown by some other part of the code depending on what this function returns and what argument was passed to `first_level_from_bids`.

Throwing may help users understand why a subsequent error happened but missing json may not be a sufficient reason to throw an error.